### ISSUE TITLE
Running unit tests  Uncaught Error: Class 'Craft' not found

### ISSUE THREAD

**Issue by Anubarak** [craftcms/cms#6843]
### Description
I'm trying to create my first unit test and I'm a bit lost currently.
I did everything mentioned [in the docs](https://craftcms.com/docs/3.x/testing/testing-craft/getting-started.html#step-1-install-codeception-3-1-or-higher). And now my files are exactly the same as in your documentation.

Trying to run `codecept run unit` (not sure where that file should come from?) results in an error so I guess that should be `./vendor/bin/codecept run unit` but that will throw an exception: 

> root@2ndold:/var/www/myspa/dev/htdocs# ./vendor/bin/codecept run unit
Codeception PHP Testing Framework v4.1.7
Powered by PHPUnit 8.5.8 by Sebastian Bergmann and contributors.
Running with seed:
> PHP Fatal error:  Uncaught Error: Class 'Craft' not found in /var/www/myspa/dev/htdocs/vendor/codeception/codeception/src/Codeception/Lib/Di.php:84
Stack trace:
#0 /var/www/myspa/dev/htdocs/vendor/codeception/codeception/src/Codeception/Lib/Di.php(84): ReflectionClass->newInstanceArgs(Array)
#1 /var/www/myspa/dev/htdocs/vendor/codeception/codeception/src/Codeception/Lib/ModuleContainer.php(124): Codeception\Lib\Di->instantiate('craft\\test\\Craf...', Array, false)
#2 /var/www/myspa/dev/htdocs/vendor/codeception/codeception/src/Codeception/SuiteManager.php(70): Codeception\Lib\ModuleContainer->create('\\craft\\test\\Cra...')
#3 /var/www/myspa/dev/htdocs/vendor/codeception/codeception/src/Codeception/Codecept.php(191): Codeception\SuiteManager->__construct(Object(Symfony\Component\EventDispatcher\EventDispatcher), 'unit', Array)
#4 /var/www/myspa/dev/htdocs/vendor/codeception/codeception/src/Codeception/Codecept.php(163): Codeception\Codecept->runSuite(Array, 'unit', NULL)
#5 /var/www/myspa/dev/htdocs/vendor/codeception/codecept in /var/www/myspa/dev/htdocs/vendor/codeception/codeception/src/Codeception/Lib/Di.php on line 84

I searched in your issues and found [#5636](https://github.com/craftcms/cms/issues/5636) so I installed `"codeception/codeception": "^4.0.0"`.

After that, I received the error `Fatal error: Class 'Codeception\Module\Yii2' not found` so I installed 
```text
composer require --dev codeception/module-yii2
```

and now the Craft not found error appears. What can I do to solve that?

### Additional info

- Craft version: Craft Pro 3.5.8
- PHP version: 7.3.21

**Comment by Anubarak** [comment]
Maybe worth noting.. when try to initiate any other class in `vendor/codeception/codeception/app.php` it fails because of the same reason

```php
echo file_exists('/path/to/project/vendor/craftcms/cms/src/test/Craft.php') // true
$plugins = new \craft\services\Plugins(); // fails in the init function -> Craft not found, but the class itself is found
$view = new \craft\web\View(); // fails in the init function -> Craft not found, but the class itself is found
$craft = new \Craft(); // fatal error, Craft not found
```

I also tried to dump my autoloader, the classes are there, I can even create an instance of `craft\test\Craft` in my main applications module but for whatever reason it is not possible during codeception. Is there any composer configuration I missed?
Even when I include the file on my own

```php
require_once __DIR__ . '/autoload.php';

use Codeception\Application;
include '/var/www/myspa/dev/htdocs/vendor/craftcms/cms/src/test/Craft.php';
$craft = new \craft\test\Craft();
```

When I try to load another class in that directory via
```PHP
require_once __DIR__ . '/autoload.php';

use Codeception\Application;
$fixture = new \craft\test\Fixture();
```
Then Yii is not found
> PHP Fatal error:  Uncaught Error: Class 'Yii' not found in /var/www/myspa/dev/htdocs/vendor/yiisoft/yii2/di/Instance.php:163

**Comment by Anubarak** [comment]
For whatever strange reasons it suddenly worked once after I created an instance of `yii\BaseYii`
Craft and Yii were found after that but I have no clue why

**Comment by brandonkelly** [comment]
Weird, perhaps your code was getting run before Composer’s autoloader was getting registered?

**Comment by Richeh** [comment]
For posterity: I just had a very similar problem that wasn't fixed by instantiating Yii; it was, however, fixed by running the tests in PHP 7.3 instead of 8.x.
